### PR TITLE
fix(cli): handle Docker SemVer build metadata in version parsing #5965

### LINKED ISSUE #5965: langgraph up fails when Docker version includes SemVer build metadata (e.g., 28.1.1+1)
### Checked other resources

- [x] This is a bug, not a usage question. For questions, please use the LangChain Forum (https://forum.langchain.com/).
- [x] I added a clear and detailed title that summarizes the issue.
- [x] I read what a minimal reproducible example is (https://stackoverflow.com/help/minimal-reproducible-example).
- [x] I included a self-contained, minimal example that demonstrates the issue INCLUDING all the relevant imports. The code run AS IS to reproduce the issue.

### Example Code

```python
"""LangGraph single-node graph template.

Returns a predefined response. Replace logic and configuration as needed.
"""

from __future__ import annotations

from dataclasses import dataclass
from typing import Any, Dict, TypedDict

from langgraph.graph import StateGraph
from langgraph.runtime import Runtime


class Context(TypedDict):
    """Context parameters for the agent.

    Set these when creating assistants OR when invoking the graph.
    See: https://langchain-ai.github.io/langgraph/cloud/how-tos/configuration_cloud/
    """

    my_configurable_param: str


@dataclass
class State:
    """Input state for the agent.

    Defines the initial structure of incoming data.
    See: https://langchain-ai.github.io/langgraph/concepts/low_level/#state
    """

    changeme: str = "example"


async def call_model(state: State, runtime: Runtime[Context]) -> Dict[str, Any]:
    """Process input and returns output.

    Can use runtime context to alter behavior.
    """
    return {
        "changeme": "output from call_model. "
        f"Configured with {runtime.context.get('my_configurable_param')}"
    }


# Define the graph
graph = (
    StateGraph(State, context_schema=Context)
    .add_node(call_model)
    .add_edge("__start__", "call_model")
    .compile(name="New Graph")
)
```

### Error Message and Stack Trace (if applicable)

```shell
Traceback (most recent call last):
  File "/home/azureuser/.local/bin/langgraph", line 8, in <module>
    sys.exit(cli())
  File "/home/azureuser/.local/lib/python3.10/site-packages/click/core.py", line 1442, in __call__
    return self.main(*args, **kwargs)
  File "/home/azureuser/.local/lib/python3.10/site-packages/click/core.py", line 1363, in main
    rv = self.invoke(ctx)
  File "/home/azureuser/.local/lib/python3.10/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/azureuser/.local/lib/python3.10/site-packages/click/core.py", line 1226, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/azureuser/.local/lib/python3.10/site-packages/click/core.py", line 794, in invoke
    return callback(*args, **kwargs)
  File "/home/azureuser/.local/lib/python3.10/site-packages/langgraph_cli/analytics.py", line 96, in decorator
    return func(*args, **kwargs)
  File "/home/azureuser/.local/lib/python3.10/site-packages/langgraph_cli/cli.py", line 223, in up
    capabilities = langgraph_cli.docker.check_capabilities(runner)
  File "/home/azureuser/.local/lib/python3.10/site-packages/langgraph_cli/docker.py", line 79, in check_capabilities
    docker_version = _parse_version(info["ServerVersion"])
  File "/home/azureuser/.local/lib/python3.10/site-packages/langgraph_cli/docker.py", line 43, in _parse_version
    return Version(int(major.lstrip("v")), int(minor), int(patch.split("-")[0]))
ValueError: invalid literal for int() with base 10: '1+1'
```

### Description

langgraph up crashes when Docker reports a SemVer with build metadata (e.g., 28.1.1+1). The CLI naively casts dot-separated segments to integers; hitting the patch 1+1 raises a ValueError. 

Reproducible with the Snap Store Docker. You can view it [here](https://snapcraft.io/docker)

### System Info

System Information
------------------
> OS:  Linux
> OS Version:  #36~22.04.1-Ubuntu SMP Tue Jul  1 03:54:01 UTC 2025
> Python Version:  3.12.11 (main, Jun  4 2025, 08:56:18) [GCC 11.4.0]

Package Information
-------------------
> langchain_core: 0.3.72
> langchain: 0.3.27
> langsmith: 0.4.13
> langchain_google_vertexai: 2.0.28
> langchain_openai: 0.3.28
> langchain_text_splitters: 0.3.9
> langgraph_cli: 0.3.6
> langgraph_sdk: 0.2.0

Optional packages not installed
-------------------------------
> langserve

Other Dependencies
------------------
> anthropic[vertexai]: Installed. No version info available.
> async-timeout<5.0.0,>=4.0.0;: Installed. No version info available.
> bottleneck: 1.5.0
> click>=8.1.7: Installed. No version info available.
> google-cloud-aiplatform: 1.109.0
> google-cloud-storage: 2.19.0
> httpx: 0.28.1
> httpx-sse: 0.4.1
> httpx<1,>=0.23.0: Installed. No version info available.
> httpx>=0.25.2: Installed. No version info available.
> jsonpatch<2.0,>=1.33: Installed. No version info available.
> langchain-anthropic;: Installed. No version info available.
> langchain-aws;: Installed. No version info available.
> langchain-azure-ai;: Installed. No version info available.
> langchain-cohere;: Installed. No version info available.
> langchain-community;: Installed. No version info available.
> langchain-core<1.0.0,>=0.3.68: Installed. No version info available.
> langchain-core<1.0.0,>=0.3.72: Installed. No version info available.
> langchain-deepseek;: Installed. No version info available.
> langchain-fireworks;: Installed. No version info available.
> langchain-google-genai;: Installed. No version info available.
> langchain-google-vertexai;: Installed. No version info available.
> langchain-groq;: Installed. No version info available.
> langchain-huggingface;: Installed. No version info available.
> langchain-mistralai: Installed. No version info available.
> langchain-mistralai;: Installed. No version info available.
> langchain-ollama;: Installed. No version info available.
> langchain-openai;: Installed. No version info available.
> langchain-perplexity;: Installed. No version info available.
> langchain-text-splitters<1.0.0,>=0.3.9: Installed. No version info available.
> langchain-together;: Installed. No version info available.
> langchain-xai;: Installed. No version info available.
> langgraph-api<0.3.0,>=0.2.67;: Installed. No version info available.
> langgraph-runtime-inmem>=0.6.0;: Installed. No version info available.
> langgraph-sdk>=0.1.0;: Installed. No version info available.
> langsmith-pyo3>=0.1.0rc2;: Installed. No version info available.
> langsmith>=0.1.17: Installed. No version info available.
> langsmith>=0.3.45: Installed. No version info available.
> numexpr: 2.11.0
> openai-agents>=0.0.3;: Installed. No version info available.
> openai<2.0.0,>=1.86.0: Installed. No version info available.
> opentelemetry-api>=1.30.0;: Installed. No version info available.
> opentelemetry-exporter-otlp-proto-http>=1.30.0;: Installed. No version info available.
> opentelemetry-sdk>=1.30.0;: Installed. No version info available.
> orjson>=3.10.1: Installed. No version info available.
> orjson>=3.9.14;: Installed. No version info available.
> packaging>=23.2: Installed. No version info available.
> pyarrow: 19.0.1
> pydantic: 2.11.7
> pydantic<3,>=1: Installed. No version info available.
> pydantic<3.0.0,>=2.7.4: Installed. No version info available.
> pydantic>=2.7.4: Installed. No version info available.
> pytest>=7.0.0;: Installed. No version info available.
> python-dotenv>=0.8.0;: Installed. No version info available.
> PyYAML>=5.3: Installed. No version info available.
> requests-toolbelt>=1.0.0: Installed. No version info available.
> requests<3,>=2: Installed. No version info available.
> requests>=2.0.0: Installed. No version info available.
> rich>=13.9.4;: Installed. No version info available.
> SQLAlchemy<3,>=1.4: Installed. No version info available.
> tenacity!=8.4.0,<10.0.0,>=8.1.0: Installed. No version info available.
> tiktoken<1,>=0.7: Installed. No version info available.
> typing-extensions>=4.7: Installed. No version info available.
> validators: 0.35.0
> vcrpy>=7.0.0;: Installed. No version info available.
> zstandard>=0.23.0: Installed. No version info available.
<details>
<summary>Agent Context</summary>
<open-swe-do-not-edit-task-plan>
{
  "tasks": [
    {
      "id": "28fcefde-9038-4a27-9aad-61d5dd224b32",
      "title": "Fix Docker version parsing to handle SemVer build metadata in langgraph CLI",
      "request": "**langgraph up fails when Docker version includes SemVer build metadata (e.g., 28.1.1+1)**\n\n### Checked other resources\n\n- [x] This is a bug, not a usage question. For questions, please use the LangChain Forum (https://forum.langchain.com/).\n- [x] I added a clear and detailed title that summarizes the issue.\n- [x] I read what a minimal reproducible example is (https://stackoverflow.com/help/minimal-reproducible-example).\n- [x] I included a self-contained, minimal example that demonstrates the issue INCLUDING all the relevant imports. The code run AS IS to reproduce the issue.\n\n### Example Code\n\n```python\n\"\"\"LangGraph single-node graph template.\n\nReturns a predefined response. Replace logic and configuration as needed.\n\"\"\"\n\nfrom __future__ import annotations\n\nfrom dataclasses import dataclass\nfrom typing import Any, Dict, TypedDict\n\nfrom langgraph.graph import StateGraph\nfrom langgraph.runtime import Runtime\n\n\nclass Context(TypedDict):\n    \"\"\"Context parameters for the agent.\n\n    Set these when creating assistants OR when invoking the graph.\n    See: https://langchain-ai.github.io/langgraph/cloud/how-tos/configuration_cloud/\n    \"\"\"\n\n    my_configurable_param: str\n\n\n@dataclass\nclass State:\n    \"\"\"Input state for the agent.\n\n    Defines the initial structure of incoming data.\n    See: https://langchain-ai.github.io/langgraph/concepts/low_level/#state\n    \"\"\"\n\n    changeme: str = \"example\"\n\n\nasync def call_model(state: State, runtime: Runtime[Context]) -> Dict[str, Any]:\n    \"\"\"Process input and returns output.\n\n    Can use runtime context to alter behavior.\n    \"\"\"\n    return {\n        \"changeme\": \"output from call_model. \"\n        f\"Configured with {runtime.context.get('my_configurable_param')}\"\n    }\n\n\n# Define the graph\ngraph = (\n    StateGraph(State, context_schema=Context)\n    .add_node(call_model)\n    .add_edge(\"__start__\", \"call_model\")\n    .compile(name=\"New Graph\")\n)\n```\n\n### Error Message and Stack Trace (if applicable)\n\n```shell\nTraceback (most recent call last):\n  File \"/home/azureuser/.local/bin/langgraph\", line 8, in <module>\n    sys.exit(cli())\n  File \"/home/azureuser/.local/lib/python3.10/site-packages/click/core.py\", line 1442, in __call__\n    return self.main(*args, **kwargs)\n  File \"/home/azureuser/.local/lib/python3.10/site-packages/click/core.py\", line 1363, in main\n    rv = self.invoke(ctx)\n  File \"/home/azureuser/.local/lib/python3.10/site-packages/click/core.py\", line 1830, in invoke\n    return _process_result(sub_ctx.command.invoke(sub_ctx))\n  File \"/home/azureuser/.local/lib/python3.10/site-packages/click/core.py\", line 1226, in invoke\n    return ctx.invoke(self.callback, **ctx.params)\n  File \"/home/azureuser/.local/lib/python3.10/site-packages/click/core.py\", line 794, in invoke\n    return callback(*args, **kwargs)\n  File \"/home/azureuser/.local/lib/python3.10/site-packages/langgraph_cli/analytics.py\", line 96, in decorator\n    return func(*args, **kwargs)\n  File \"/home/azureuser/.local/lib/python3.10/site-packages/langgraph_cli/cli.py\", line 223, in up\n    capabilities = langgraph_cli.docker.check_capabilities(runner)\n  File \"/home/azureuser/.local/lib/python3.10/site-packages/langgraph_cli/docker.py\", line 79, in check_capabilities\n    docker_version = _parse_version(info[\"ServerVersion\"])\n  File \"/home/azureuser/.local/lib/python3.10/site-packages/langgraph_cli/docker.py\", line 43, in _parse_version\n    return Version(int(major.lstrip(\"v\")), int(minor), int(patch.split(\"-\")[0]))\nValueError: invalid literal for int() with base 10: '1+1'\n```\n\n### Description\n\nlanggraph up crashes when Docker reports a SemVer with build metadata (e.g., 28.1.1+1). The CLI naively casts dot-separated segments to integers; hitting the patch 1+1 raises a ValueError. \n\nReproducible with the Snap Store Docker. You can view it [here](https://snapcraft.io/docker)\n\n### System Info\n\nSystem Information\n------------------\n> OS:  Linux\n> OS Version:  #36~22.04.1-Ubuntu SMP Tue Jul  1 03:54:01 UTC 2025\n> Python Version:  3.12.11 (main, Jun  4 2025, 08:56:18) [GCC 11.4.0]\n\nPackage Information\n-------------------\n> langchain_core: 0.3.72\n> langchain: 0.3.27\n> langsmith: 0.4.13\n> langchain_google_vertexai: 2.0.28\n> langchain_openai: 0.3.28\n> langchain_text_splitters: 0.3.9\n> langgraph_cli: 0.3.6\n> langgraph_sdk: 0.2.0\n\nOptional packages not installed\n-------------------------------\n> langserve\n\nOther Dependencies\n------------------\n> anthropic[vertexai]: Installed. No version info available.\n> async-timeout<5.0.0,>=4.0.0;: Installed. No version info available.\n> bottleneck: 1.5.0\n> click>=8.1.7: Installed. No version info available.\n> google-cloud-aiplatform: 1.109.0\n> google-cloud-storage: 2.19.0\n> httpx: 0.28.1\n> httpx-sse: 0.4.1\n> httpx<1,>=0.23.0: Installed. No version info available.\n> httpx>=0.25.2: Installed. No version info available.\n> jsonpatch<2.0,>=1.33: Installed. No version info available.\n> langchain-anthropic;: Installed. No version info available.\n> langchain-aws;: Installed. No version info available.\n> langchain-azure-ai;: Installed. No version info available.\n> langchain-cohere;: Installed. No version info available.\n> langchain-community;: Installed. No version info available.\n> langchain-core<1.0.0,>=0.3.68: Installed. No version info available.\n> langchain-core<1.0.0,>=0.3.72: Installed. No version info available.\n> langchain-deepseek;: Installed. No version info available.\n> langchain-fireworks;: Installed. No version info available.\n> langchain-google-genai;: Installed. No version info available.\n> langchain-google-vertexai;: Installed. No version info available.\n> langchain-groq;: Installed. No version info available.\n> langchain-huggingface;: Installed. No version info available.\n> langchain-mistralai: Installed. No version info available.\n> langchain-mistralai;: Installed. No version info available.\n> langchain-ollama;: Installed. No version info available.\n> langchain-openai;: Installed. No version info available.\n> langchain-perplexity;: Installed. No version info available.\n> langchain-text-splitters<1.0.0,>=0.3.9: Installed. No version info available.\n> langchain-together;: Installed. No version info available.\n> langchain-xai;: Installed. No version info available.\n> langgraph-api<0.3.0,>=0.2.67;: Installed. No version info available.\n> langgraph-runtime-inmem>=0.6.0;: Installed. No version info available.\n> langgraph-sdk>=0.1.0;: Installed. No version info available.\n> langsmith-pyo3>=0.1.0rc2;: Installed. No version info available.\n> langsmith>=0.1.17: Installed. No version info available.\n> langsmith>=0.3.45: Installed. No version info available.\n> numexpr: 2.11.0\n> openai-agents>=0.0.3;: Installed. No version info available.\n> openai<2.0.0,>=1.86.0: Installed. No version info available.\n> opentelemetry-api>=1.30.0;: Installed. No version info available.\n> opentelemetry-exporter-otlp-proto-http>=1.30.0;: Installed. No version info available.\n> opentelemetry-sdk>=1.30.0;: Installed. No version info available.\n> orjson>=3.10.1: Installed. No version info available.\n> orjson>=3.9.14;: Installed. No version info available.\n> packaging>=23.2: Installed. No version info available.\n> pyarrow: 19.0.1\n> pydantic: 2.11.7\n> pydantic<3,>=1: Installed. No version info available.\n> pydantic<3.0.0,>=2.7.4: Installed. No version info available.\n> pydantic>=2.7.4: Installed. No version info available.\n> pytest>=7.0.0;: Installed. No version info available.\n> python-dotenv>=0.8.0;: Installed. No version info available.\n> PyYAML>=5.3: Installed. No version info available.\n> requests-toolbelt>=1.0.0: Installed. No version info available.\n> requests<3,>=2: Installed. No version info available.\n> requests>=2.0.0: Installed. No version info available.\n> rich>=13.9.4;: Installed. No version info available.\n> SQLAlchemy<3,>=1.4: Installed. No version info available.\n> tenacity!=8.4.0,<10.0.0,>=8.1.0: Installed. No version info available.\n> tiktoken<1,>=0.7: Installed. No version info available.\n> typing-extensions>=4.7: Installed. No version info available.\n> validators: 0.35.0\n> vcrpy>=7.0.0;: Installed. No version info available.\n> zstandard>=0.23.0: Installed. No version info available.",
      "completed": true,
      "createdAt": 1755674970827,
      "taskIndex": 0,
      "planRevisions": [
        {
          "plans": [
            {
              "plan": "**Fix the `_parse_version` function in `libs/cli/langgraph_cli/docker.py`** to handle SemVer build metadata by modifying line 43 from `int(patch.split('-')[0])` to `int(patch.split('-')[0].split('+')[0])`. This change will properly extract the numeric patch version from versions like '28.1.1+1' by splitting on both '-' (prerelease) and '+' (build metadata) characters.",
              "index": 0,
              "completed": true,
              "summary": "**Successfully fixed the `_parse_version` function in `libs/cli/langgraph_cli/docker.py` to handle SemVer build metadata**\n\n### Actions Taken:\n- **Located the problematic code**: Found the issue in line 43 of the `_parse_version` function where `int(patch.split(\"-\")[0])` was failing when the patch version contained build metadata (e.g., \"1+1\" from version \"28.1.1+1\")\n- **Applied the fix**: Modified line 43 from `int(patch.split(\"-\")[0])` to `int(patch.split(\"-\")[0].split(\"+\")[0])` to properly handle both prerelease (`-`) and build metadata (`+`) separators\n- **Verified the fix**: Created and ran a comprehensive test script that confirmed the fix works correctly for all version formats including:\n  - Normal versions: `28.1.1` → `(28, 1, 1)`\n  - Build metadata versions: `28.1.1+1` → `(28, 1, 1)` (the originally failing case)\n  - Prerelease versions: `1.2.3-alpha` → `(1, 2, 3)`\n  - Combined versions: `1.2.3-alpha+build` → `(1, 2, 3)`\n  - Versions with 'v' prefix: `v1.2.3` → `(1, 2, 3)`\n  - Edge cases: `1.2` → `(1, 2, 0)` and `1` → `(1, 0, 0)`\n\n### Technical Details:\n- The fix uses chained split operations to extract only the numeric part of the patch version\n- This approach maintains backward compatibility with existing version formats\n- The solution is minimal and focused, avoiding unnecessary complexity\n- All test cases pass, confirming the fix resolves the original ValueError while preserving existing functionality\n\n### Impact:\n- Resolves the CLI failure when Docker reports versions with SemVer build metadata\n- Ensures the `langgraph up` command works correctly with Docker installations that use build metadata in version strings\n- Maintains proper version comparison functionality for Docker capability detection"
            },
            {
              "plan": "**Add comprehensive tests for the `_parse_version` function in `libs/cli/tests/unit_tests/test_docker.py`** by importing `_parse_version` from `langgraph_cli.docker` and creating test cases covering: normal versions ('1.2.3'), versions with 'v' prefix ('v1.2.3'), prerelease versions ('1.2.3-alpha'), build metadata versions ('1.2.3+1'), combined prerelease and build metadata ('1.2.3-alpha+build'), and edge cases with missing components ('1.2', '1').",
              "index": 1,
              "completed": true,
              "summary": "**Successfully added comprehensive tests for the `_parse_version` function in `libs/cli/tests/unit_tests/test_docker.py`**\n\n### Actions Taken:\n- **Added import**: Modified the import statement to include `_parse_version` from `langgraph_cli.docker`\n- **Created 6 comprehensive test functions** covering all required scenarios:\n  1. `test_parse_version_normal_versions()` - Tests normal semantic versions like `1.2.3`, `28.1.1`, `0.0.1`\n  2. `test_parse_version_with_v_prefix()` - Tests versions with 'v' prefix like `v1.2.3`, `v28.1.1`\n  3. `test_parse_version_prerelease()` - Tests prerelease versions like `1.2.3-alpha`, `1.2.3-beta.1`, `1.2.3-rc.1`\n  4. `test_parse_version_build_metadata()` - Tests build metadata versions like `1.2.3+1`, `28.1.1+1` (the originally failing case), `1.2.3+build.1`\n  5. `test_parse_version_combined_prerelease_and_build()` - Tests combined versions like `1.2.3-alpha+1`, `1.2.3-beta.1+build.1`\n  6. `test_parse_version_edge_cases()` - Tests edge cases with missing components like `1.2`, `1`, `v1.2`, `v1`\n\n### Test Results:\n- **All 6 new test functions pass** (100% success rate)\n- Tests specifically validate the fix for the original issue (`28.1.1+1` → `Version(28, 1, 1)`)\n- Edge cases are properly handled with default values for missing components\n- Tests ensure backward compatibility with existing version formats\n\n### Test Coverage:\n- **Normal semantic versions**: Validates standard three-part versions\n- **'v' prefix handling**: Ensures the `lstrip(\"v\")` functionality works correctly\n- **Prerelease versions**: Confirms `-` separator handling extracts numeric patch\n- **Build metadata versions**: Validates `+` separator handling (the core fix)\n- **Combined formats**: Tests complex versions with both prerelease and build metadata\n- **Edge cases**: Ensures proper defaults for incomplete version strings\n\n### Technical Quality:\n- Tests are well-organized with descriptive function names and docstrings\n- Each test function focuses on a specific aspect of version parsing\n- Test cases include both common and edge case scenarios\n- Comments highlight the originally failing case for future reference"
            },
            {
              "plan": "**Run the CLI test suite** using `make test` in the `libs/cli` directory to ensure the fix doesn't break existing functionality and all tests pass, then run `make format` and `make lint` to ensure code quality standards are met.",
              "index": 2,
              "completed": true,
              "summary": "**Successfully ran the CLI test suite and code quality checks**\n\n### Actions Taken:\n\n#### Test Suite Execution:\n- **Ran `make test`** in the `libs/cli` directory\n- **Docker-specific tests**: All 14 tests in `test_docker.py` **PASSED** (100% success rate)\n- **New `_parse_version` tests**: All 6 new test functions **PASSED**, confirming the fix works correctly\n- **Overall test results**: 60 out of 65 tests passed\n- **Test failures**: 5 failures were identified, but all are **unrelated to the `_parse_version` fix**\n\n#### Test Failure Analysis:\nThe 5 failing tests are all related to Docker not being installed in the test environment:\n- `test_dockerfile_command_with_docker_compose`\n- `test_build_command_shows_wolfi_warning` \n- `test_build_generate_proper_build_context`\n- `test_build_command_with_api_version`\n- `test_build_command_with_api_version_and_base_image`\n\nAll failures show the same error: `\"Error: Docker not installed\"`, confirming these are environment-related issues, not code defects.\n\n#### Code Quality Checks:\n- **`make format`**: ✅ **PASSED** - Reformatted 2 files, 36 files left unchanged\n- **`make lint`**: ✅ **PASSED** - All ruff checks passed, no formatting issues, no import issues, mypy checks completed\n\n#### Verification of Fix:\n- **Core functionality**: The `_parse_version` function now correctly handles SemVer build metadata\n- **Test coverage**: Comprehensive tests validate all version formats including the originally failing case (`28.1.1+1`)\n- **Backward compatibility**: All existing docker tests continue to pass\n- **Code quality**: No linting or formatting issues introduced\n\n### Key Insights:\n- The fix successfully resolves the CLI failure with Docker versions containing build metadata\n- The failing tests are environmental (Docker not installed) and don't impact the core functionality\n- All code quality standards are maintained\n- The solution is minimal, robust, and well-tested"
            }
          ],
          "createdAt": 1755674970827,
          "createdBy": "agent",
          "revisionIndex": 0
        }
      ],
      "activeRevisionIndex": 0,
      "pullRequestNumber": 5966,
      "completedAt": 1755675342060,
      "summary": "I've successfully fixed the `_parse_version` function in the LangGraph CLI to handle SemVer build metadata by modifying the version parsing logic. Comprehensive tests were added to cover all version formats, and all code quality checks passed. The fix resolves the issue with Docker versions containing build metadata while maintaining backward compatibility."
    }
  ],
  "activeTaskIndex": 0
}
</open-swe-do-not-edit-task-plan>
</details>

### PR DESCRIPTION
**Pull request by sarathak**
Description:
Corrects _parse_version to support Docker versions with SemVer build metadata (e.g., 28.1.1+1), resolving #5965. Adds comprehensive unit tests for version parsing, including normal, v-prefixed, prerelease, build metadata, combined prerelease/build metadata, and edge cases with missing components.

Issue:
Closes #5965

Dependencies:
None

### CODE REVIEW

File: libs/cli/tests/unit_tests/test_docker.py [hunk 1]
```diff
@@ -363,3 +366,28 @@ def test_compose_with_api_version_and_debugger():
...
+    ],
+)
+def test__parse_version_comprehensive(input_str, expected):
+    """
+    Test the _parse_version function with comprehensive input cases.
+    """
+    result = _parse_version(input_str)
+    assert result == expected, (
+        f"Failed for {input_str}: got {result}, expected {expected}"
+    )
```
Comment (sydney-runkle) [inline libs/cli/tests/unit_tests/test_docker.py]:
```suggestion
    assert _parse_version(input_str) == expected
```

File: libs/cli/tests/unit_tests/test_docker.py [hunk 2]
```diff
@@ -363,3 +366,28 @@ def test_compose_with_api_version_and_debugger():
...
+        ("1", Version(1, 0, 0)),
+        ("v28.1.1+1", Version(28, 1, 1)),
+        ("2.0.0-beta.1+exp.sha.5114f85", Version(2, 0, 0)),
+        ("v3.4.5-rc1+build.123", Version(3, 4, 5)),
+    ],
+)
+def test__parse_version_comprehensive(input_str, expected):
+    """
+    Test the _parse_version function with comprehensive input cases.
+    """
```
Comment (sydney-runkle) [inline libs/cli/tests/unit_tests/test_docker.py]:
```suggestion
def test_parse_version_w_edge_cases(input_str, expected):
```

File: libs/cli/langgraph_cli/docker.py [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import json
 import pathlib
+import re
```
Comment (sydney-runkle) [inline libs/cli/langgraph_cli/docker.py]: Would prefer to not have extra import

Comment (sydney-runkle) [inline libs/cli/langgraph_cli/docker.py]:
```suggestion
```

File: libs/cli/langgraph_cli/docker.py [hunk 4]
```diff
@@ -40,7 +41,9 @@ def _parse_version(version: str) -> Version:
         patch = "0"
     else:
         major, minor, patch = parts
-    return Version(int(major.lstrip("v")), int(minor), int(patch.split("-")[0]))
+    return Version(
+        int(major.lstrip("v")), int(minor), int(re.split("[\-\+]", patch)[0])
```
Comment (sydney-runkle) [inline libs/cli/langgraph_cli/docker.py]:
```suggestion
        int(major.lstrip("v")), int(minor), int(patch.split("-")[0].split("+")[0])
```
